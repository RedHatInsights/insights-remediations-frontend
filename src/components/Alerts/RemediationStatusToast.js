import React, { useState, useEffect } from 'react';
import propTypes from 'prop-types';

import { Alert } from '@patternfly/react-core';

function RemediationStatusToast ({ status, name, ...props }) {

    const [ isVisible, setisVisible ] = useState(true);

    useEffect(() => {
        setTimeout(() => setisVisible(false), 8000);
    });

    return (
        <React.Fragment>
            { isVisible &&
                <Alert
                    variant={ (status === 'passed' ? 'success' : 'danger') }
                    title={ `Remediation plan ${ name } ${ status === 'passed' ? 'successfully completed' : 'failed' }` }
                    { ...props }
                />
            }
        </React.Fragment>
    );
};

export default RemediationStatusToast;

RemediationStatusToast.propTypes = {
    status: propTypes.oneOf([ 'passed', 'failed' ]),
    name: propTypes.string.isRequired
};

RemediationStatusToast.defaultProps = {
    status: 'passed'
};
