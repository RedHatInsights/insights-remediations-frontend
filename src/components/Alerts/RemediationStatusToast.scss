.ins-c-remediation-toast__status {
    position: absolute;
    right:  var(--pf-c-page__main-section--PaddingRight);
    top: var(--pf-c-page__header--MinHeight);
    margin-top: var(--pf-c-page__main-section--PaddingTop);
    z-index: 10;
    max-width: 600px;
    word-wrap: anywhere;
}