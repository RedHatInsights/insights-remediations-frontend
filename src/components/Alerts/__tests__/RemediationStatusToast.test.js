import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import RemediationStatusToast from '../RemediationStatusToast';

describe('ExecuteBanner component', () => {

    it('should render', () => {
        const wrapper = mount(<RemediationStatusToast name='passed remediation'/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render failed', () => {
        const wrapper = mount(<RemediationStatusToast name='failed remediation' status='failed'/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
