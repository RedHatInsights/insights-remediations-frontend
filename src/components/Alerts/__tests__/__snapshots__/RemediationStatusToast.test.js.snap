// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExecuteBanner component should render 1`] = `
<RemediationStatusToast
  name="passed remediation"
  status="passed"
>
  <Component
    title="Remediation plan passed remediation successfully completed"
    variant="success"
  >
    <ComponentWithOuia
      component={[Function]}
      componentProps={
        Object {
          "title": "Remediation plan passed remediation successfully completed",
          "variant": "success",
        }
      }
      consumerContext={null}
    >
      <Alert
        ouiaContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
        title="Remediation plan passed remediation successfully completed"
        variant="success"
      >
        <div
          aria-label="Success Alert"
          className="pf-c-alert pf-m-success"
        >
          <AlertIcon
            variant="success"
          >
            <div
              className="pf-c-alert__icon"
            >
              <CheckCircleIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    transform=""
                  />
                </svg>
              </CheckCircleIcon>
            </div>
          </AlertIcon>
          <h4
            className="pf-c-alert__title"
          >
            <span
              className="pf-u-screen-reader"
            >
              Success alert:
            </span>
            Remediation plan passed remediation successfully completed
          </h4>
        </div>
      </Alert>
    </ComponentWithOuia>
  </Component>
</RemediationStatusToast>
`;

exports[`ExecuteBanner component should render failed 1`] = `
<RemediationStatusToast
  name="failed remediation"
  status="failed"
>
  <Component
    title="Remediation plan failed remediation failed"
    variant="danger"
  >
    <ComponentWithOuia
      component={[Function]}
      componentProps={
        Object {
          "title": "Remediation plan failed remediation failed",
          "variant": "danger",
        }
      }
      consumerContext={null}
    >
      <Alert
        ouiaContext={
          Object {
            "isOuia": false,
            "ouiaId": null,
          }
        }
        title="Remediation plan failed remediation failed"
        variant="danger"
      >
        <div
          aria-label="Danger Alert"
          className="pf-c-alert pf-m-danger"
        >
          <AlertIcon
            variant="danger"
          >
            <div
              className="pf-c-alert__icon"
            >
              <ExclamationCircleIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
                title={null}
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    transform=""
                  />
                </svg>
              </ExclamationCircleIcon>
            </div>
          </AlertIcon>
          <h4
            className="pf-c-alert__title"
          >
            <span
              className="pf-u-screen-reader"
            >
              Danger alert:
            </span>
            Remediation plan failed remediation failed
          </h4>
        </div>
      </Alert>
    </ComponentWithOuia>
  </Component>
</RemediationStatusToast>
`;
